cms_form/static/src/js/ajax.js

```diff
@@ -0,0 +1,80 @@
+odoo.define('cms_form.ajax', function (require) {
+    'use strict';
+
+    var core = require('web.core'),
+        animation = require('website.content.snippets.animation');
+
+    animation.registry.CMSFormAjax = animation.Class.extend({
+        selector: ".cms_form_wrapper form[data-ajax]",
+        start: function () {
+            var data = this.$el.data('form');
+            this.$el.on('submit', this.proxy('submit_form'));
+            if (data.ajax_onchange) {
+                this.$el.on('change', this.proxy('submit_form'));
+            };
+            jQuery(data.form_content_selector).find(
+                '.pagination a[href]'
+            ).on(
+                'click', this.proxy('pager')
+            );
+        },
+        ajax_submit: function (data, post_data) {
+            return jQuery.ajax(
+                _.str.sprintf('/cms/ajax/search/%s', data.model),
+                {
+                    data: post_data,
+                    dataType: 'json',
+                    success: this.proxy('success'),
+                    error: this.proxy('error'),
+                }
+            );
+        },
+        submit_form: function (ev) {
+            var $this = this.$el,
+                data = $this.data('form'),
+                $container = jQuery(data.form_content_selector);
+
+            jQuery.blockUI();
+            ev.preventDefault();
+            $container.empty();
+
+            return this.ajax_submit(
+                data, $this.serialize() + '&csrf_token=' + core.csrf_token
+            );
+        },
+        success: function (data) {
+            jQuery.unblockUI();
+            var $container = jQuery(
+                    this.$el.data('form').form_content_selector
+                );
+            $container.html(data.content);
+            $container.find('.pagination a[href]').on(
+                'click', this.proxy('pager')
+            );
+        },
+        error: function (data) {
+            jQuery.unblockUI();
+        },
+        pager: function (ev) {
+            var $this = this.$el,
+                data = $this.data('form'),
+                $a = jQuery(ev.currentTarget);
+
+            jQuery.blockUI();
+            ev.preventDefault();
+
+            return this.ajax_submit(
+                data,
+                $a.prop('search').substring(1).replace(/&page=[^&]+/, '') +
```

Comment:
having `data-page` attribute on the pager links would make this easier... but this is another story :)